test/mongodb.test.js

```diff
@@ -2652,6 +2677,22 @@ describe('mongodb connector', function() {
     });
   });
 
+  it('should allow to find using case insensitive index', function(done) {
+    Category.create({title: 'My Category'}, function(err, category1) {
+      should.not.exist(err);
```

Comment:
Sorry, let's change this assertion to `if (err) done(err);` (see @bajtos' comment [here](https://github.com/strongloop/loopback-datasource-juggler/pull/1597#discussion_r215872053) for more context).